apple/internal/apple_framework_import.bzl

```diff
@@ -386,14 +535,42 @@ def _apple_static_framework_import_impl(ctx):
         transitive_sets = transitive_sets,
     ))
 
-    framework_groups = _grouped_framework_files(framework_imports)
-    framework_binaries = _all_framework_binaries(framework_groups)
+    is_framework = False
+    for f in framework_imports:
+        if f.dirname.endswith(".framework"):
+            is_framework = True
+            break
 
-    objc_provider_fields = _framework_objc_provider_fields(
-        "static_framework_file",
-        module_map_imports,
-        framework_binaries,
-    )
+    if is_framework:
+        framework_groups = _grouped_framework_files(framework_imports)
+        framework_binaries = _all_framework_binaries(
+            frameworks_groups = framework_groups,
+        )
+
+        objc_provider_fields = _framework_objc_provider_fields(
+            "static_framework_file",
+            module_map_imports,
+            framework_binaries,
+        )
+    else:
+        framework_groups = _grouped_xcframework_files(framework_imports)
+        framework_binaries = []
+
+        # For non-framework types (XCFrameworks not embedding any .framework
+        # bundle but only contain static libraries, headers, and module maps),
+        # assume the library filename is the same with XCFramework name or has
+        # the .a extension. If the library file has a different naming, the
+        # XCFramework can't be processed now.
+        for f in other_imports:
+            file_basename = f.basename
+            if file_basename == framework_name or file_basename.endswith(".a"):
+                framework_binaries.append(f)
```

Comment:
should we error if this ends up being empty? seems like even if that did work we should discourage it unless there's a real need, since it's more likely an accident

Reply:
There's already a check for that in the original implementation. Apparently when `alwayslink = False`, it's acceptable to not have binaries.

https://github.com/bazelbuild/rules_apple/blob/6dbfd9e0d741be9f6a1b68759db6f1ab1de0a2d1/apple/internal/apple_framework_import.bzl#L398-L400

Reply:
maybe better to not keep that behavior for this new rule tho? seems a bit strange